spec/compiler/codegen/macro_spec.cr

```diff
@@ -1895,4 +1895,34 @@ describe "Code gen: macro" do
       {% end %}
       )).to_i.should eq(10)
   end
+
+  it "returns an array of each call" do
+    run(%(
+      macro test
+        {{@caller.size == 1 ? 1 : 0}}
+      end
+
+      test
+      )).to_i.should eq(1)
+  end
```

Comment:
**suggestion:** Move these specs out of `codegen` and refactor with `assert_type`.
The feature should be completely semantical, so there's no need to actually compile any code, just typing it should be sufficient.
For example:
```suggestion
  it "returns an array of each call" do
    assert_type(<<-CRYSTAL) { int32 }
      macro test
        {{ @caller.size == 1 ? 1 : 'a' }}
      end

      test
      CRYSTAL
  end
```
This applies to all specs here. I suppose they could go into `semantic/macro_spec.cr`.